software/base/src/main/java/org/apache/brooklyn/entity/software/base/lifecycle/MachineLifecycleEffectorTasks.java

```diff
@@ -444,9 +493,6 @@ public void run() {
                 }
             }
             entity().addLocations(ImmutableList.of((Location) machine));
-            if (entity().getAttribute(Attributes.JCLOUDS_PROVISIONING_RUNNING) != null) {
```

Comment:
One big reason that someone might call stop() while it's still starting is they realise their provisioning parameters were wrong, so want to abort. Therefore continuing to support that use-case seems important.